server/src/main/java/org/opensearch/accesscontrol/resources/ShareWith.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.accesscontrol.resources;
+
+import org.opensearch.core.xcontent.ToXContentFragment;
+import org.opensearch.core.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.List;
+
+/**
+ * This class contains information about whom a resource is shared with.
+ * It could be a user-name, a role or a backend_role.
+ *
+ * @opensearch.experimental
+ */
+public class ShareWith implements ToXContentFragment {
+
+    private List<String> users;
```

Comment:
I'm hesitant to introduce notions of roles, backend roles or user into the core. I'm wondering if Security can provide a class similar to this to plugins (similar to SPI model from JS), but not have plugins formally extending the security plugin. If a plugin extends another plugin then it means that the other plugin must be installed in a cluster before the plugin that depends on another can be installed.

With resource sharing, if the security plugin is not installed then all users will have full access to all plugin resources.

With the security plugin installed it can enforce authz on plugin resources using this new mechanism.